edxval/tests/test_api.py

```diff
@@ -2035,3 +2035,87 @@ def test_create_transcript_preferences(self):
 
         # Verify that there should be 2 preferences exists
         self.assertEqual(TranscriptPreference.objects.count(), 2)
+
+
+@ddt
+class TranscripCredentialsStateTest(TestCase):
+    """
+    OrganizationTranscriptCredentialsState Tests
+    """
+    def setUp(self):
+        """
+        Tests setup
+        """
+        OrganizationTranscriptCredentialsState.objects.create(
+            org='edX', provider='Cielo24', exists=False
+        )
+        OrganizationTranscriptCredentialsState.objects.create(
+            org='edX', provider='3PlayMedia', exists=False
+        )
+
+    def assert_credential_state():
+        """
+        Verify credential state.
+        """
+        pass
+
+    @data(
+        {'provider': ''},
+        {'provider': 'fake'},
+        {'provider': None},
+    )
+    @unpack
+    def test_provider_validation(self, provider):
+        """
+        Verify that provider value is validated
+        """
+        with self.assertRaises(ValidationError):
+            OrganizationTranscriptCredentialsState.objects.create(
+                org='edX', provider=provider, exists=False
+            )
+
+    @data(
+        {'org': 'MAX', 'provider': 'Cielo24', 'exists': True},
+        {'org': 'MAX', 'provider': '3PlayMedia', 'exists': True},
+    )
+    @unpack
+    def test_credentials_state_update(self, **kwargs):
+        """
+        Verify that `update_organization_transcript_credentials_state` method works as expected
+        """
+        api.update_organization_transcript_credentials_state(
+            org=kwargs['org'], provider=kwargs['provider'], exists=kwargs['exists']
+        )
+
+        credentials_state = OrganizationTranscriptCredentialsState.objects.get(org=kwargs['org'])
+        for key in kwargs:
+            self.assertEqual(getattr(credentials_state, key), kwargs[key])
+
+    @data(
+        {'org': 'edX', 'provider': 'Cielo24'},
+        {'org': 'edX', 'provider': '3PlayMedia'},
+    )
+    @unpack
+    def test_get_credentials_state(self, org, provider):
```

Comment:
We should also add a test if org does not match, what is expected result we should expect